schemas/v1.0/azure.yaml.json

```diff
@@ -1,8 +1,71 @@
 {
-    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$schema": "https://json-schema.org/draft/2019-09/schema",
     "$id": "https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/v1.0/azure.yaml.json",
     "type": "object",
-    "required": ["name", "services"],
+    "required": [
+        "name",
+        "services"
+    ],
+    "definitions": {
+        "hooks": {
+            "title": "Definition of hooks for developer integration",
+            "description": "Hooks should match `azd` command names or `service` event names prefixed with `pre` or `post` depending on when the script should execute. When specifying paths they should be relative to the project or service path.",
+            "additionalProperties": {
+                "$ref": "#/definitions/hook"
+            }
+        },
+        "hook": {
+            "type": "object",
+            "title": "Definition of hooks for developer integration",
+            "description": "Hooks should match `azd` command names or `service` event names prefixed with `pre` or `post` depending on when the script should execute relative to the command",
+            "minProperties": 1,
+            "additionalProperties": {
+                "type": "object",
+                "additionalProperties": false,
+                "properties": {
+                    "shell": {
+                        "type": "string",
+                        "title": "Type of shell to execute scripts",
+                        "description": "Optional. The type of shell to use for the hook. (Default: sh)",
+                        "enum": [
+                            "sh",
+                            "pwsh"
+                        ],
+                        "default": "sh"
+                    },
+                    "run": {
+                        "type": "string",
+                        "title": "Required. The inline script or relative path of your scripts from the project or service path",
+                        "description": "When specifying an inline script you also must specify the `shell` to use. This is automatically inferred when using paths."
+                    },
+                    "continueOnError": {
+                        "type": "boolean",
+                        "default": false,
+                        "title": "Whether or not a script error will halt the azd command",
+                        "description": "Optional. When set to true will continue to run the command even after a script error has occurred. (Default: false)"
+                    },
+                    "interactive": {
+                        "type": "boolean",
+                        "default": true,
+                        "title": "Whether the script will run in interactive mode",
+                        "description": "Optional. When set to true will bind the script to stdin, stdout & stderr of the running console. (Default: false)"
```

Comment:
Noticed that the description is saying default to `false`, but the default is actually set to `true`. Do you know which one we want?

Reply:
The default is `false`.  It will always write out to the console that the hook ran, just won't write out any output of the hook calls themselves unless `interactive` is set to true.